### PR TITLE
Add 64-bit support

### PR DESCRIPTION
**Pull request by kenr**
Previously the serialization did not support 64-bit architectures on the app side.

This PR adds this support by adding a different buffer handling mechanism.

Other fixes:
- rssi_get response decoding in SDv3

### CODE REVIEW

File: src/common/app_ble_gap.cpp [hunk 1]
```diff
@@ -455,4 +456,50 @@ uint32_t app_ble_gap_adv_set_unregister(uint8_t adv_handle, uint8_t **pp_adv_dat
...
+
+    catch (const std::out_of_range &)
+    {
+        return 0;
+    }
+}
+
+void *app_ble_gap_adv_buf_unregister(const int id)
+{
+    if (!app_ble_gap_check_current_adapter_set(EVENT_CODEC_CONTEXT))
```
Comment (kenr) [inline src/common/app_ble_gap.cpp]: Can we ensure that this function is only called in EVENT_CODEC_CONTEXT?

File: src/common/app_ble_gap.cpp [hunk 2]
```diff
@@ -455,4 +456,50 @@ uint32_t app_ble_gap_adv_set_unregister(uint8_t adv_handle, uint8_t **pp_adv_dat
         return NRF_ERROR_SD_RPC_INVALID_STATE;
     }
 }
+
+int app_ble_gap_adv_buf_register(void * p_buf)
+{
+    if (!app_ble_gap_check_current_adapter_set(REQUEST_REPLY_CODEC_CONTEXT))
```
Comment (kenr) [inline src/common/app_ble_gap.cpp]: Can we ensure that this function is only called REQUEST_REPLY_CODEC_CONTEXT?

Comment (nordic-krch) [reply src/common/app_ble_gap.cpp]: i think that is true.

File: src/common/app_ble_gap.cpp (hunk 1 above)
Comment (nordic-krch) [reply src/common/app_ble_gap.cpp]: that may not be true. eg it should be unregistered if `scan_start` returns error (it may not be done now)

Comment (nordic-krch) [reply src/common/app_ble_gap.cpp]: started to think about `scan_stop` case. Will there be any event that will unregister the buffer? If not then this case is not handled.

**Comment by nordic-krch** [comment]
so now my main concerns are:
- scan_start returning error
- adv_set_configure returning error
- scan_stop

In those cases application and connectivity should clean up registered buffers. I think this should go in and fix for that can go in another round.

File: include/common/internal/app_ble_gap.h [hunk 3]
```diff
@@ -216,7 +216,25 @@ uint32_t app_ble_gap_adv_set_register(uint8_t adv_handle, uint8_t *p_adv_data,
...
+ * @brief Register an advertisement buffer pointer
+ * 
+ * @param[in] p_buf Advertisement buffer to create a pointer ID from
+ * @return -1 if there is no space left in buffer table, 0 of p_buf is nullptr, >0 with buffer location in buffer table
+ */
+
 int app_ble_gap_adv_buf_register(void * p_buf);
+
+/**
+ * @breif Unregister a buffer from advertisement buffer table
```
Comment (bencefr) [inline include/common/internal/app_ble_gap.h]: typo s/breif/brief/

**Pull request merged**